web/api/v1/api.go

```diff
@@ -1567,6 +1610,47 @@ func parseExcludeAlerts(r *http.Request) (bool, error) {
 	return excludeAlerts, nil
 }
 
+func parseListRulesPaginationRequest(r *http.Request) (*listRulesPaginationRequest, *parsePaginationError) {
+	var (
+		parsedMaxGroups int64 = -1
+		err             error
+	)
+	maxGroups := r.URL.Query().Get("max_groups")
+	nextToken := r.URL.Query().Get("next_token")
```

Comment:
`next_token` is great if we will never paginate over rules (: Are we sure about this? Otherwise maybe `group_next_token` and `group_limit` would be a nice, explicit pair?